src/hotspot/share/gc/parallel/psAdaptiveSizePolicy.cpp

```diff
@@ -73,14 +71,28 @@ void PSAdaptiveSizePolicy::print_stats(bool is_survivor_overflowing) {
     is_survivor_overflowing ? "true" : "false");
 }
 
+// The throughput goal is implemented as
+//      _throughput_goal = 1 - (1 / (1 + gc_cost_ratio))
+// gc_cost_ratio is the ratio
+//      application cost / gc cost
+// For example a gc_cost_ratio of 4 translates into a
+// throughput goal of .80
+static double calculate_throughput_goal(double gc_cost_ratio) {
+  return 1.0 - (1.0 / (1.0 + gc_cost_ratio));
+}
+
 size_t PSAdaptiveSizePolicy::compute_desired_eden_size(bool is_survivor_overflowing, size_t cur_eden) {
   // Guard against divide-by-zero; 0.001ms
   double gc_distance = MAX2(_gc_distance_seconds_seq.last(), 0.000001);
   double min_gc_distance = MinGCDistanceSecond;
 
-  if (mutator_time_percent() < _throughput_goal) {
+  // Get a local copy and use it inside gc-pause in case the global var gets updated externally.
+  const uint local_GCTimeRatio = GCTimeRatio;
```

Comment:
needs an `Atomic::load()` as the compiler is free to re-read the global later at any use of the local.

Reply:
I was thinking to add that when this flag actually becomes manageable. I don't feel too strongly about it. Do you prefer using atomic already now?

Reply:
I prefer to either do this with that change then, or have the change complete. Separating the main change out seems prone for errors, and no different than not having this patch in.

Reply:
Using atomic op now. All Parallel-specific changes are in place.